### ISSUE TITLE
How to use the parsed SQL with Prisma

### ISSUE THREAD

**Issue by gj1118** [react-querybuilder/react-querybuilder#414]
Hey Guys, 
I am using react-querybuilder to query the DB. I have set it up and can format the query to sql. But I am trying to use the formatted query with prisma ORM. Apparently I am unaware as to how the parameterized query will work. 

Can you please help me out with this please?  

Another way, I was thinking was to have a representation of the selected data in the UI . I already have it as I need to populate my data table. Is there a way I can filter the JSON array using the generated formatted query.

**Comment by jakeboone02** [comment]
Hi @gj1118. I haven't used Prisma ORM before, but it looks like the [`where` syntax](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#where) is similar to [JsonLogic](https://jsonlogic.com/), which this library can [import](https://react-querybuilder.js.org/docs/api/import#jsonlogic) and [export](https://react-querybuilder.js.org/docs/api/export#jsonlogic). It probably wouldn't be much of a stretch to alter the JsonLogic export to produce a Prisma-compatible JSON object. Let me know if that's what you need and I'll investigate further.

Speaking of JsonLogic, I think that might be the answer to your second question about filtering the data on the client. See [json-logic-js](https://www.npmjs.com/package/json-logic-js).

**Comment by jakeboone02** [comment]
I just saw the Prisma [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) method which takes a SQL template string. Is that what you're using? In my previous comment I assumed you were using [`findMany`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findmany).

**Comment by gj1118** [comment]
> 

Thank you so much for your reply. Much appreciated. I had indeed used queryRaw. The problem with queryRaw is that in order to safeguard against sql inject attacks it parameterizes the input query. so anything of this sort will fail `select * from posts where author = 1`. I can use queryRayUnsafe and it will indeed work , but then it increases the chances of sql injection attacks. 

I did perform a little test with `json-logic-js` and so far, all tests are green. 👍🏽 

Thank you once again for looking at an issue on a Friday evening. I hope you have a good rest of the day. Thanks once again. 

With Warm Regards, 
Gagan
